via/static/scripts/video_player/components/ToastMessages.tsx

```diff
@@ -0,0 +1,190 @@
+import type { ComponentChildren } from 'preact';
+import { useCallback, useRef, useState } from 'preact/hooks';
+
+export type ToastMessage = {
+  id: string;
+  type: 'error' | 'success' | 'notice';
+  message: ComponentChildren;
+
+  /**
+   * Visually hidden messages are announced to screen readers but not visible.
+   * Defaults to false.
+   */
+  visuallyHidden?: boolean;
+
+  /**
+   * Determines if the toast message should be auto-dismissed.
+   * Defaults to true.
+   */
+  autoDismiss?: boolean;
+};
+
+type ToastMessageItemProps = {
+  message: ToastMessage;
+  onDismiss: (id: string) => void;
+};
+
+/**
+ * An individual toast message: a brief and transient success or error message.
+ * The message may be dismissed by clicking on it. `visuallyHidden` toast
+ * messages will not be visible but are still available to screen readers.
+ */
+function ToastMessageItem({ message, onDismiss }: ToastMessageItemProps) {
+  // Capitalize the message type for prepending; Don't prepend a message
+  // type for "notice" messages
+  const prefix =
+    message.type !== 'notice'
+      ? `${message.type.charAt(0).toUpperCase() + message.type.slice(1)}: `
+      : '';
+
+  return (
+    <Callout
+      classes={classnames({
+        'sr-only': message.visuallyHidden,
+      })}
+      status={message.type}
+      onClick={() => onDismiss(message.id)}
+      variant="raised"
+    >
+      <strong>{prefix}</strong>
+      {message.message}
+    </Callout>
+  );
+}
+
+const BaseToastMessageTransition: TransitionComponent = ({
+  direction = 'in',
+  onTransitionEnd,
+  children,
+}) => {
+  const isDismissed = direction === 'out';
+  const containerRef = useRef<HTMLDivElement>(null);
+  const handleAnimation = (e: AnimationEvent) => {
+    // Ignore non-relevant animations:
+    // * Happening on children elements
+    // * Animations which are not relevant for toast messages getting "in" or "out"
+    if (
+      e.target !== containerRef.current ||
+      !['slide-in-from-right', 'fade-in', 'fade-out'].includes(e.animationName)
+    ) {
+      // We cannot test this scenario. It will be more relevant when/if this
+      // component is extracted at some point
+      /* istanbul ignore next */
+      return;
+    }
+
+    onTransitionEnd?.(direction);
+  };
+
+  return (
+    <div
+      onAnimationEnd={handleAnimation}
+      ref={containerRef}
+      className={classnames('relative w-full container', {
+        // Only ever slide-from-right if motion-safe is preferred
+        'lg:motion-safe:animate-slide-in-from-right animate-fade-in':
+          !isDismissed,
+        // Only ever fade-in if motion-reduction is preferred
+        'motion-reduce:animate-fade-in': !isDismissed,
+        'animate-fade-out': isDismissed,
+      })}
+    >
+      {children}
+    </div>
+  );
+};
+
+export type ToastMessagesProps = {
+  messages: ToastMessage[];
+  onMessageDismiss: (id: string) => void;
+  _setTimeout?: typeof setTimeout;
```

Comment:
```suggestion
  setTimeout_?: typeof setTimeout;
```

It is more conventional in our code to put the underscore for this kind of seam at the end of the name rather than the beginning (see `window_` and `document_` in the client).

Reply:
Oh, right. I think I got it mixed with the prefix for private props 😅